### PR TITLE
Merge pull request #1350 from philanthropy-u/mabdullahz/LP-1729-v2

### PR DESCRIPTION
**Pull request by mabdullahz**

### CODE REVIEW

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 1]
```diff
@@ -198,7 +197,7 @@ def create_account_with_params(request, params):
     # Announce registration
     REGISTER_USER.send(sender=None, user=user, registration=registration)
 
-    create_comments_service_user(user)
+    # create_comments_service_user(user)
```
Comment (saad-waqar) [inline openedx/core/djangoapps/user_authn/views/register.py]: lets add a comment here as to why we have commented this code.

File: openedx/features/student_account/helpers.py [hunk 2]
```diff
@@ -46,3 +55,78 @@ def get_non_active_course(user):
...
+
+            user_organization.save()
+
+            user_extended_profile_data = {
+                "organization_id": user_organization.id
+            }
+
+    # create User Extended Profile
+    user_extended_profile = UserExtendedProfile.objects.create(user=user, **user_extended_profile_data)
+    post_save.send(UserExtendedProfile, instance=user_extended_profile, created=False)
```
Comment (ahmed-faraz46) [inline openedx/features/student_account/helpers.py]: Let's please add a comment here to clarify, why we need to dispatch this signal manually, why not the signal dispatched by upper create worked here?

File: openedx/features/student_account/helpers.py [hunk 3]
```diff
@@ -46,3 +55,78 @@ def get_non_active_course(user):
...
+
+    # create user email preferences object
+    EmailPreference.objects.create(user=user, opt_in=form.cleaned_data.get('opt_in'))
```
Comment (ahmed-faraz46) [inline openedx/features/student_account/helpers.py]: Let's please justify the creation of email preferences in the name of this method as well.

File: openedx/core/djangoapps/user_authn/views/register.py (hunk 1 above)
Comment (mabdullahz) [reply openedx/core/djangoapps/user_authn/views/register.py]: Done

File: openedx/features/student_account/helpers.py (hunk 2 above)
Comment (mabdullahz) [reply openedx/features/student_account/helpers.py]: Done

File: openedx/features/student_account/helpers.py (hunk 3 above)
Comment (mabdullahz) [reply openedx/features/student_account/helpers.py]: Done

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 4]
```diff
@@ -180,7 +181,8 @@ def create_account_with_params(request, params):
     if skip_email:
         registration.activate()
     else:
-        compose_and_send_activation_email(user, profile, registration)
+        data = get_params_for_activation_email(request, registration, user)
```
Comment (ahmed-faraz46) [inline openedx/core/djangoapps/user_authn/views/register.py]: Let's please change this method `get_params_for_activation_email` to prepare data and send the email as well, In this way we can reduce the core change.

**Pull request merged**